InstanceManager.js

```diff
@@ -409,94 +425,171 @@ class InstanceManager {
   }
 
-  startCluster(numAgents, numCoordinators, numDbServers, options = {}) {
+  async startCluster(numAgents, numCoordinators, numDbServers, options = {}) {
     let agencyOptions = options.agents || {};
     _.extend(agencyOptions, {agencySize: numAgents});
 
-    return this.startAgency(agencyOptions)
-      .then(agents => { return new Promise.delay(2000, agents); })
-      .then(agents => {
-        let promises = [Promise.resolve(agents)];
+    debugLog(`Starting cluster A:${numAgents} C:${numCoordinators} D:${numDbServers}`);
+    debugLog("booting agents...");
 
-        let dbServers = Array.from(Array(numDbServers).keys()).map(index => {
-          return this.startDbServer('dbServer-' + (index + 1));
-        });
-        return Promise.all(dbServers);
-      })
-      .then(dbServers => { return new Promise.delay(2000, dbServers); })
-      .then(dbServers => {
-        let coordinators = Array.from(
-          Array(numCoordinators).keys()
-        ).map(index => {
-          return this.startCoordinator('coordinator-' + (index + 1));
-        });
-        return Promise.all([...coordinators, ...dbServers]);
-      })
-      .then(servers => {
-        return this.waitForAllInstances();
-      })
-      .then(() => {
-        return this.getEndpoint();
-      });
+    let agents = await this.startAgency(agencyOptions);
+    this.waitForInstances(this.agents());
+    debugLog("all agents are booted");
+
+    await sleep(2000);
+    let dbServers = Array.from(Array(numDbServers).keys()).map(index => {
+      return this.startDbServer('dbServer-' + (index + 1));
+    });
+
+    debugLog("booting DBServers...");
+    await Promise.all(dbServers);
+    this.waitForInstances(this.dbServers());
+    debugLog("all DBServers are booted");
+    await sleep(2000);
+
+    let coordinators = Array.from(
+      Array(numCoordinators).keys()
+    ).map(index => {
+      return this.startCoordinator('coordinator-' + (index + 1));
+    });
+    debugLog("booting Coordinators...");
+    await Promise.all(coordinators);
+    debugLog("all Coordinators are booted");
+
+    debugLog("waiting for /_api/version to succeed an all servers");
+    await this.waitForAllInstances();
+    debugLog("Cluster is up and running");
+
+    return this.getEndpoint();
   }
 
   async waitForInstance(instance, started = Date.now()) {
-    if (instance.status !== 'RUNNING') {
-      throw new Error(`Instance ${instance.name} is down!`);
-    }
-    if (Date.now() - started > WAIT_TIMEOUT * 1000) {
-      throw new Error(
-        `Instance ${instance.name} is still not ready after ${WAIT_TIMEOUT} secs`
-      );
+    while (true) {
+      if (instance.status !== 'RUNNING') {
+        throw new Error(`
+          Instance ${instance.name} is down!
+          Real status ${instance.status} with exitcode ${instance.exitcode}.
+          See logfile here ${instance.logFile}`);
+      }
+     
+      let _test_ = Date.now() - started;
```

Comment:
What's with the underlines? Any reason not to inline this value into the `if` check? Otherwise if it isn't reassigned it should probably be const and have a meaningful name without underlines (e.g. `elapsedTime`).